conan/tools/microsoft/toolchain.py

```diff
@@ -187,3 +207,16 @@ def _write_main_toolchain(self, config_filename, condition):
         conan_toolchain = "\n".join(line for line in conan_toolchain.splitlines() if line.strip())
         self._conanfile.output.info("MSBuildToolchain writing {}".format(self.filename))
         save(main_toolchain_path, conan_toolchain)
+
+    def _get_extra_flags(self):
+        # Now, it's time to get all the flags defined by the user
+        cxxflags = self._conanfile.conf.get("tools.build:cxxflags", default=[], check_type=list)
+        cflags = self._conanfile.conf.get("tools.build:cflags", default=[], check_type=list)
+        sharedlinkflags = self._conanfile.conf.get("tools.build:sharedlinkflags", default=[], check_type=list)
+        exelinkflags = self._conanfile.conf.get("tools.build:exelinkflags", default=[], check_type=list)
+        defines = self._conanfile.conf.get("tools.build:defines", default=[], check_type=list)
+        return {
+            "compiler_flags": cflags + cxxflags,
```

Comment:
that also probably needs some sort of check/filter. it might be useless (if not breaking) to apply C++ specific flags to pure C targets.

Reply:
`MSBuildDeps` is already doing this sum, I only copied and paste the same logic.

Reply:
well, it might be a serious bug, which we missed because `MSBuildDeps` is stiill not used so widely (in CCI at least).
if it's too much to fix in this PR, let's create an another issue then :)

Reply:
Here is the same. I'm not able to get any clue about how to split both flags into different sections. Keeping it as it is.